File: leetcode/jump-game.js
// https://leetcode.com/problems/jump-game

/**
 * @param {number[]} nums
 * @return {boolean}
 */
let canJump = function (nums) {
  let reachable = 0;
  for (let i = 0; i <= reachable; ++i) {
    reachable = Math.max(reachable, i + nums[i]);
    if (reachable >= nums.length - 1) {
      return true;
    }
  }
  return false;
};

File: leetcode/jump-game.cpp
// https://leetcode.com/problems/jump-game

class Solution {
public:
  bool canJump(vector<int> &nums) {
    int reachable = 0;
    for (int i = 0; i <= reachable; ++i) {
      reachable = std::max(reachable, i + nums[i]);
      if (reachable >= nums.size() - 1) {
        return true;
      }
    }
    return false;
  }
};